src/connection.ts

```diff
@@ -1902,6 +1902,9 @@ class Connection extends EventEmitter {
 
     if (this.config.options.port) {
       return this.connectOnPort(this.config.options.port, this.config.options.multiSubnetFailover, signal, this.config.options.connector);
+    } else if (this.config.options.connector) {
+      // port and multiSubnetFailover are not used when using a custom connector
+      return this.connectOnPort(0, false, signal, this.config.options.connector);
```

Comment:
There is a possibility that user proved both port and connector from config, the logic will go into the if statement instead of the else if and still work properly since the connector is passed into connectOnPort. Another case is, if use only pass in connector and logic will got into the else if and still call the connectOnPort. 
Under both cases, with or without port passed into connectOnPort function, the connect socket here will be provided by the passed in customConnector, and the passed in port number should be ignored within connectOnPort.

Reply:
thanks @MichaelSun90! these are also great examples of the kind of tests I can add to make codecov happy 😊

Reply:
I added the connector check to take precedence just to err on the safe side but at the same time I made sure to set `port` to undefined during the validation phase of the constructor along with adding tests for the combinations of providing port/server/connector. Let me know if it looks better now 😊